src/tdastro/astro_utils/passbands.py

```diff
@@ -106,6 +107,29 @@ def _update_waves(self) -> None:
         else:
             self.waves = np.unique(np.concatenate([passband.waves for passband in self.passbands.values()]))
 
+    def _calculate_in_band_wave_indices(self) -> None:
+        """Calculate the indices of the group's wave grid that are in the passband's wave grid.
+
+        Eg, if a group's waves are [11, 12, 13, 14, 15] and a single band's are [13, 14], we get [2, 3].
+
+        The indices are stored in the passband's _in_band_wave_indices attribute as either a tuple of two ints
+        (lower, upper) or a 1D np.ndarray of ints.
+        """
+        for passband in self.passbands.values():
+            # We only want the fluxes that are in the passband's wavelength range
+            # So, find the indices in the group's wave grid that are in the passband's wave grid
+            lower, upper = passband.waves[0], passband.waves[-1]
+            lower_index, upper_index = np.searchsorted(self.waves, [lower, upper])
+
+            # Check that this is the right grid after all (check will fail if passbands overlap and passbands
+            # do not happen to be on the same phase of the grid; eg, even if the step is 10, if the first
+            # passband starts at 100 and the second at 105, the passbands won't share the same grid)
+            if np.array_equal(self.waves[lower_index : upper_index + 1], passband.waves):
+                indices = (lower_index, upper_index + 1)
+            else:
+                indices = np.searchsorted(self.waves, passband.waves)
```

Comment:
In the original code, you had a `np.unique` call here. Is that no longer needed?

Reply:
Yes--the non-unique values I had been seeing were a result of me having gotten the arguments to `np.searchsorted` flipped (was previously `(passband.waves, self.waves)`).

But as long as all elements of `passband.waves` are included in `self.waves`, and neither of them contain duplicate/non-unique values, all elements of the returned array should already be unique.